frontend/src/metabase/visualizations/visualizations/Sankey.jsx

```diff
@@ -0,0 +1,330 @@
+/* eslint-disable react/prop-types */
+import React, { Component } from "react";
+import { t } from "ttag";
+import d3 from "d3";
+import cx from "classnames";
+
+import _ from "underscore";
+
+import { sankey, sankeyLinkHorizontal } from "d3-sankey";
+import chroma from "chroma-js";
+
+import {
+  columnsAreValid,
+  preserveExistingColumnsOrder,
+} from "metabase/visualizations/lib/utils";
+import { columnSettings } from "metabase/visualizations/lib/settings/column";
+
+import { getOptionFromColumn } from "metabase/visualizations/lib/settings/utils";
+import { getDefaultColumns } from "../lib/settings/graph";
+
+const PADDING_BOTTOM = 10;
+
+export default class Sankey extends Component {
+  static uiName = t`Sankey`;
+  static identifier = "sankey";
+  static iconName = "sankey";
+
+  static minSize = { width: 4, height: 4 };
+
+  static isSensible({ cols, rows }) {
+    return rows.length >= 1 && cols.length >= 3;
+  }
+
+  static checkRenderable(series, settings) {
+    const start = settings["sankey.start"];
+    const destinations = settings["sankey.destinations"];
+    for (let i = 0; i < destinations.length; i++) {
+      if (destinations[i] === start[0]) {
+        throw new Error(t`Start can't be a destination.`);
+      }
+    }
+  }
+
+  static settings = {
+    ...columnSettings({
+      getColumns: ([
+        {
+          data: { cols },
+        },
+      ]) => cols,
+      hidden: true,
```

Comment:
It was really hard to test this chart, I can't even try to render the chart, but I believe if `hidden` is true then you shouldn't even need to set `getColumns` as this settings won't be visible anyway.

The `columnSettings` will show **Formatting** tab, but from [this comment](https://github.com/metabase/metabase/pull/26976#pullrequestreview-1238194772), it doesn't seem to have **Formatting** tab at all.